### PR TITLE
add mapi_machinehealthcheck_short_circuit metric

### PR DESCRIPTION
**Pull request by elmiko**
This change adds the counter metric to track when a MachineHealthCheck
has been short-circuited. It adds helper functions for enabling and
disabling the metric, and an update to the metrics doc showing the new
metric.

### CODE REVIEW

**Comment by elmiko** [comment]
creating this as a draft while i do some testing
it will need to go after #754

**Comment by elmiko** [comment]
this is ready for review but still needs to go after the other pr
/hold on #754 being merged

**Comment by elmiko** [comment]
updated to add the namespace label and change the Register... func call to Observe...

**Comment by elmiko** [comment]
not sure why this is causing conflicts, once #752 and #754 have merged i will rebase

**Comment by elmiko** [comment]
rebased

**Comment by elmiko** [comment]
rebased after merge of 754
/hold cancel

**Review by michaelgugino — CHANGES_REQUESTED** [review]
Are we sure we want to call it 'short_circuit' as that is a little idiomatic.  Is 'short circuit' described elsewhere?  We use "RemediationAllowed" as the condition, perhaps we should use that for the metric and reverse the boolean?  Or "TooManyUnhealthyNodes" to match the reason and keep the boolean the same?

If our docs say 'short circuit' I suppose that's fine, but we should try to be consistent if not.

**Review by michaelgugino — APPROVED** [review]
/lgtm

**Review by JoelSpeed — COMMENTED** [review]
/approve

**Pull request merged**
